examples/nlp/training/trainer.py

```diff
@@ -0,0 +1,141 @@
+﻿#  Copyright (c) maiot GmbH 2021. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+
+import os
+import tensorflow as tf
+from typing import List, Text, Dict
+from zenml.core.steps.trainer.tensorflow_trainers.tf_base_trainer import \
+    TFBaseTrainerStep
+from transformers import TFDistilBertForSequenceClassification
+from zenml.utils.post_training.post_training_utils import \
+    get_feature_spec_from_schema
+
+
+class UrduTrainer(TFBaseTrainerStep):
+    def __init__(self,
+                 model_name: Text,
+                 serving_model_dir: Text = None,
+                 transform_output: Text = None,
+                 train_files: List[Text] = None,
+                 eval_files: List[Text] = None,
+                 batch_size: int = 64,
+                 epochs: int = 25,
+                 learning_rate: float = 1e-4,
+                 schema_file: Text = None,
+                 **kwargs
+                 ):
+
+        super(UrduTrainer, self).__init__(model_name=model_name,
+                                          serving_model_dir=serving_model_dir,
+                                          transform_output=transform_output,
+                                          train_files=train_files,
+                                          eval_files=eval_files,
+                                          batch_size=batch_size,
+                                          epochs=epochs,
+                                          learning_rate=learning_rate,
+                                          **kwargs)
+
+        # ML variables
+        self.batch_size = batch_size
+        self.epochs = epochs
+        self.learning_rate = learning_rate
+        self.model_name = model_name
+
+        if schema_file:
+            self.schema_path = os.path.dirname(schema_file)
+        else:
+            self.schema_path = None
+        self.schema_file = schema_file
+
+    def get_run_fn(self):
+        return self.run_fn
+
+    def run_fn(self):
+        feature_spec = get_feature_spec_from_schema(self.schema_path)
+        train_dataset = self.input_fn(self.train_files, feature_spec)
+        eval_dataset = self.input_fn(self.eval_files, feature_spec)
+
+        model = self.model_fn(train_dataset=train_dataset,
+                              eval_dataset=eval_dataset)
+
+        model.save_pretrained(self.serving_model_dir, saved_model=True)
+
+    def model_fn(self,
```

Comment:
Shouldnt we be using a HuggingFace model to showcase support there? Becasue we discussed how to convert the TFDataset into a HuggingFace dataset